gdal/port/cpl_conv.cpp

```diff
@@ -2165,7 +2165,10 @@ static double proj_strtod(char *nptr, char **endptr)
     // Scan for characters which cause problems with VC++ strtod().
     while( (c = *cp) != '\0')
     {
-        if( c == 'd' || c == 'D' )
+        if( c == 'd' || c == 'D' ||
+            // UTF-8 "DEGREE SIGN" is 0xC2 0xB0 
+            c == static_cast<char>(0xC2) ||
```

Comment:
More portable / less subjects to undefined/unspecified behaviour
```suggestion
            static_cast<unsigned char>(c) == 0xC2 ||
```